src/main/kotlin/edu/ics499/team1/app/repositories/TripRepository.kt

```diff
@@ -1,8 +1,18 @@
 package edu.ics499.team1.app.repositories
 
-import edu.ics499.team1.app.domains.Trip
+import edu.ics499.team1.app.entities.TripEntity
+import edu.ics499.team1.app.entities.VehicleEntity
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Modifying
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
 
-interface TripRepository : JpaRepository<Trip, Long> {
-  
+// custom queries by https://www.baeldung.com/spring-data-jpa-modifying-annotation
+interface TripRepository : JpaRepository<TripEntity, Int> {
+
+    @Modifying
+    @Query("update TripEntity set mileage = :mileage where tripId = :tripId")
+    fun updateMileage(@Param("tripId") tripId: Int, @Param("mileage") mileage: Int)
+
+    fun findAllByVehicle(vehicle: VehicleEntity): List<TripEntity>
```

Comment:
This is a derived query. This weeks video the teacher posted (D6) goes over this a little bit, but also just google. But we can write basic queries via these methods and spring data will know how to implement them based of field names and given values.